src/core/cities-core.js

```diff
@@ -0,0 +1,34 @@
+const {knex} = require('../util/database').connect();
+
+const getCities = (opts) => {
+  let sqlString = `
+    SELECT
+      id,
+      name
+    FROM cities
+  `;
+  let params = [];
+  let whereClauses = [];
+
+  if (opts.cityId) {
+    whereClauses.push(`id = ?`);
+    params.push(opts.cityId);
+  }
+
+  if (opts.cityName) {
+    whereClauses.push(`name = ?`);
+    params.push(opts.cityName);
+  }
+
+  if (whereClauses.length > 0) {
+    sqlString += ` WHERE ${ whereClauses.join(' AND ')}`;
+  }
+
+  return knex.raw(sqlString, params)
+    .then(result => result)
```

Comment:
It should be the core layer's responsibility to format the query result to a nice format for its consumers, i.e. return an array of city objects. In this case it's just a simple

```js
.then(result => result.rows)
```